compiler/rustc_parse/src/parser/ty.rs

```diff
@@ -833,12 +833,9 @@ impl<'a> Parser<'a> {
         let lo = self.token.span;
         let leading_token = self.prev_token.clone();
         let has_parens = self.eat(&token::OpenDelim(Delimiter::Parenthesis));
-        let inner_lo = self.token.span;
 
-        let modifiers = self.parse_trait_bound_modifiers()?;
         let bound = if self.token.is_lifetime() {
-            self.error_lt_bound_with_modifiers(modifiers);
-            self.parse_generic_lt_bound(lo, inner_lo, has_parens)?
+            self.parse_generic_lt_bound(lo, has_parens)?
         } else if self.eat_keyword(kw::Use) {
```

Comment:
Bounds like `?use<>`, `async use<>`, `const use<>` are no longer syntactically well-formed. That's actually great because `fn f() -> impl Sized + const use<> {}` (…) passes compilation on nightly, lol^^.

Seems like this warrants a regression test.

Reply:
ugh i will fix `?use<>` lmao